nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,466 @@
+                } else {
+                    responseFlowFile = session.importFrom(responseBodyStream, responseFlowFile);
+                    Optional<String> mimeType = httpResponseEntity.headers().getFirstHeader(HEADER_CONTENT_TYPE);
+                    if (mimeType.isPresent()) {
+                        responseFlowFile = session.putAttribute(responseFlowFile, CoreAttributes.MIME_TYPE.key(), mimeType.get());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            session.remove(responseFlowFile);
+            throw e;
+        }
+        return responseFlowFile;
+    }
+
+    private String createAuthorizationHeader(ProcessContext context, FlowFile flowfile) {
+        String userName = context.getProperty(WORKDAY_USERNAME).evaluateAttributeExpressions(flowfile).getValue();
+        String password = context.getProperty(WORKDAY_PASSWORD).evaluateAttributeExpressions(flowfile).getValue();
+        String base64Credential = Base64.getEncoder().encodeToString((userName + USERNAME_PASSWORD_SEPARATOR + password).getBytes(StandardCharsets.UTF_8));
+        return BASIC_PREFIX + base64Credential;
+    }
+
+    private TransformResult transformRecords(ProcessSession session, FlowFile flowfile, FlowFile responseFlowFile, String hashingAlgorithm, Set<String> columnsToHash,
+        InputStream responseBodyStream) throws IOException, SchemaNotFoundException, MalformedRecordException {
+        int numberOfRecords = 0;
+        String mimeType = null;
+        try (RecordReader reader = recordReaderFactoryReference.get().createRecordReader(flowfile,
+            new BufferedInputStream(responseBodyStream), getLogger())) {
+            RecordSchema schema = recordSetWriterFactoryReference.get()
+                .getSchema(flowfile == null ? Collections.emptyMap() : flowfile.getAttributes(), reader.getSchema());
+            try (OutputStream responseStream = session.write(responseFlowFile);
+                RecordSetWriter recordSetWriter = recordSetWriterFactoryReference.get().createWriter(getLogger(), schema, responseStream, responseFlowFile)) {
+                mimeType = recordSetWriter.getMimeType();
+                recordSetWriter.beginRecordSet();
+                Record currentRecord;
+                // as the report can be changed independently from the flow, it's safer to ignore field types and unknown fields in the Record Reading process
+                while ((currentRecord = reader.nextRecord(false, true)) != null) {
+                    for (String recordPath : columnsToHash) {
+                        RecordPathResult evaluate = RecordPath.compile("hash(" + recordPath + ", '" + hashingAlgorithm + "')").evaluate(currentRecord);
+                        evaluate.getSelectedFields().forEach(fieldVal -> fieldVal.updateValue(fieldVal.getValue(), RecordFieldType.STRING.getDataType()));
+                    }
```

Comment:
Taking a closer look at this implementation, what is the reason for supporting this hash operation? Compiling and running the record path evaluation could take time for larger record sets. It seems like it would be better to remove this feature, which could be performed if desired in a downstream processor.

Reply:
This is an optional functionality so I think it should be acceptable. Would it help if I create a Map with the compiled paths?

Reply:
Precompiling the paths might help, but it would still be much better to remove the functionality, and the associated dependency, since `nifi-record-path` brings in a number of transitive dependencies that are otherwise unnecessary for this processor.  Removing this functionality would keep the capabilities of this processor more focused on retrieval and formatting.

Reply:
I've removed the hashing functionality

Reply:
Thanks!